### DISCUSSION TITLE
Error with an if/else statment in a trigger

### DISCUSSION THREAD

**Discussion by mendlero** [supabase/supabase#29025] [Questions] [answered]
I have a table rooms in the public schema as such:
```sql
create table if not exists rooms(
    id          bigint primary KEY GENERATED ALWAYS AS IDENTITY,
    project_id  bigint references public.projects ON DELETE CASCADE NOT NULL,
    name        text not null
);
```
Every time the room is updated (like changing the name) I would like to updated the parent project accordingly (change the `updated_at` field), for that I have a function called `handle_project_update(bigint)` in the public schema.

In order to achive that I want to create a trigger function that takes care of it so Ii tried to run this code that defines a trigger function:
```sql
create or replace function public.handle_rooms_update()
    returns trigger
    language plpgsql
    set search_path = ''
    as $$
        BEGIN
            IF TG_OP = 'DELETE'
            THEN
                public.handle_project_update(OLD.project_id);
            ELSE
                public.handle_project_update(NEW.project_id);
            END IF;

            return null;
        END;
    $$;
```

but got this error:
```
ERROR:  42601: syntax error at or near "public"
LINE 9:                 public.handle_project_update(OLD.project_id);
```

and I cannot figure out what the problem is, does anyone know how to solve this error?

**Comment by gc-ft** [comment, marked as answer]
you need to run the function somehow...

Use PERFORM infront of the name.
Also, this is a trigger, do NOT return null, return NEW, unless TG_OP is DELETE then it should return OLD...

```
IF TG_OP = 'DELETE'
THEN
  PERFORM public.handle_project_update(OLD.project_id);
  RETURN OLD;
ELSE
  PERFORM public.handle_project_update(NEW.project_id);
  RETURN NEW;
END IF;
```

return null might work if you use AFTER UPDATE but if you ever use this BEFORE UPDATE you would destroy all your updates.

Fabian

**Reply by mendlero** [reply]
Thanks for the help, using `perform` solved the problem.
btw, I am returning null because it is an `after` trigger as fit for this situation thus the return value is ignored.

**Reply by gc-ft** [reply]
Great, just wanted to be sure this is not missed.
